guides/ci/travis.md

```diff
@@ -0,0 +1,232 @@
+# Travis CI
+
+If you are new to continuous integration (or you want to refresh the basic concepts) we may start reading the [core concepts guide](https://docs.travis-ci.com/user/for-beginners/) and then a [short tutorial](https://docs.travis-ci.com/user/tutorial/).
+
+Now let's see an example of using Travis CI with Crystal.
+
+## Build and run specs
+
+### Using `latest` and `nightly`
+
+When [creating a Crystal project](../../using_the_compiler/#creating-a-crystal-project) using `crystal init`, Crystal creates a `.travis.yml` file for us:
+
+```yaml
+# .travis.yml
+language: crystal
+
+crystal:
+  - latest
+  - nightly
+
+script:
+  - crystal spec
+  - crystal tool format --check
+```
+
+Although a basic configuration file, it will let us use Travis CI. Now, let's go to Travis CI dashboard to [add the GitHub repository](https://docs.travis-ci.com/user/tutorial/#to-get-started-with-travis-ci). From this moment, Travis CI will run the `specs` against the Crystal compiler (using both `latest` and `nightly` releases).
+
+### Using a specific Crystal release
+
+So, let's suppose we want to use version [0.31.1](https://github.com/crystal-lang/crystal/releases/tag/0.31.1) of the Crystal compiler. For this we are going to use [Docker](https://www.docker.com/). First we need to add Docker as a service in `.travis.yml`, and then we can use `docker` commands in our build steps, like this:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+script:
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+**Note:** A list with the different official [Crystal docker images](https://hub.docker.com/r/crystallang/crystal/tags) is available at [DockerHub](https://hub.docker.com/r/crystallang/crystal).
+
+### Using `latest`, `nightly` and a specific Crystal release all together!
+
+For using 3 versions of the compiler we are going to use a different configuration file based on [Build Matrix](https://docs.travis-ci.com/user/customizing-the-build#build-matrix)
+Here is the example:
+
+```yaml
+# .travis.yml
+matrix:
+ include:
+   - language: crystal
+     crystal:
+       - latest
+     script:
+       - crystal spec
+
+   - language: crystal
+     crystal:
+       - nightly
+     script:
+       - crystal spec
+
+   - language: bash
+     services:
+       - docker
+     script:
+       - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+## Installing shards packages
+
+Thanks to the `language: crystal` support, Travis CI automatically runs `shards install`. And for faster ci times we may use [caching](#caching)
+
```

Comment:
We should add:

When using a docker image you will need to run `shards install` explicitly. 

```
script:
  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 shards install
  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
```

Note: Since the shards will be installed in `./lib/` folder, it will be preserved for the second `docker run` command.